### ISSUE TITLE
black formatting and pylinting

### ISSUE THREAD

**Issue by mikey0000** [mikey0000/Luba-API#6]
project needs to run these on each PR and be setup.

**Comment by mikey0000** [comment]
This is partially being done with Ruff at the moment, but welcome improvements.

proto folder should be ignored.

**Comment by jbowen116** [comment]
Hi @mikey0000 — happy to take a swing at this if you're still interested. Current state from a quick survey:

**Enforced on PRs** (`.github/workflows/on-push.yml`): `ruff check`, `ruff format --check`, pytest (non-`live`), proto codegen drift.

**Configured locally but not enforced on PRs:**
- `ty` — runs in `.pre-commit-config.yaml` and per `CLAUDE.md`. Currently shows 230 diagnostics on `main` (ty is alpha and still tightening rules).
- `pylint` — referenced in `CLAUDE.md` but no pre-commit hook and no CI step. Surfaces a real reimport bug at `pymammotion/mammotion/commands/messages/navigation.py:972`, plus the usual style/positional-arg warnings.
- `pre-commit` itself — no CI runner, so `--no-verify` bypasses autotyping + ty.

A few directions, in roughly increasing lift:

1. Add [`pre-commit/action@v3.0.1`](https://github.com/pre-commit/action) so the existing pre-commit config is gated on every PR. Tightens what's already configured, no new tools.
2. Add `ty` as an informational (non-blocking) CI step so type drift is visible without blocking PRs while the tool stabilizes.
3. Add `pylint` as an informational CI step + a `tool.pylint` config section in `pyproject.toml` that excludes `pymammotion/proto/*` per your earlier comment.
4. The actual cleanup — chip away at the ty/pylint backlogs so steps 2/3 can flip to blocking.

Any of these match what you had in mind, or do you want to keep this scoped to formatting only and close the issue? Happy to open whichever PR you'd prefer.

**Comment by mikey0000** [comment]
Yeah it's about time to clean up and enforce it, it would be good to only fail the build on changed files for now, as otherwise it would be a monumental effort, the rules et in pyproject probably needs reviewed as well to relax some of the stricter errors. Go for it!
